aws/resource_aws_lb_target_group.go

```diff
@@ -282,39 +289,9 @@ func resourceAwsLbTargetGroupCreate(d *schema.ResourceData, meta interface{}) er
 		if _, ok := d.GetOk("vpc_id"); !ok {
 			return fmt.Errorf("vpc_id should be set when target type is %s", d.Get("target_type").(string))
 		}
-
 		params.Port = aws.Int64(int64(d.Get("port").(int)))
 		params.Protocol = aws.String(d.Get("protocol").(string))
 		params.VpcId = aws.String(d.Get("vpc_id").(string))
-
-		stickinessBlocks := d.Get("stickiness").([]interface{})
-		if len(stickinessBlocks) > 0 {
-			stickiness := stickinessBlocks[0].(map[string]interface{})
-
-			if d.Get("protocol").(string) == elbv2.ProtocolEnumHttp ||
-				d.Get("protocol").(string) == elbv2.ProtocolEnumHttps {
-				if stickiness["type"].(string) != "lb_cookie" {
-					return fmt.Errorf("stickiness type can only be \"lb_cookie\" when protocol is %s", d.Get("protocol").(string))
-				}
-			} else if d.Get("protocol").(string) == elbv2.ProtocolEnumTcp ||
-				d.Get("protocol").(string) == elbv2.ProtocolEnumUdp ||
-				d.Get("protocol").(string) == elbv2.ProtocolEnumTcpUdp {
-				if stickiness["type"].(string) != "source_ip" {
-					return fmt.Errorf("stickiness type can only be \"source_ip\" when protocol is %s", d.Get("protocol").(string))
-				}
-			}
-		} else {
-			if d.Get("protocol").(string) == elbv2.ProtocolEnumTcp ||
```

Comment:
By removing this `else` block, how are we ensuring that if the `stickiness` block isn't defined, the right type is used?

Reply:
I expect the `TestAccAWSLBTargetGroup_stickinessDefaultNLB` test to fail.

Reply:
I see it passing, but I don't understand why.

Reply:
The API will set the stickiness and set enabled to false.